beets/dbcore/types.py

```diff
@@ -35,7 +36,7 @@ class Type:
     """The `Query` subclass to be used when querying the field.
     """
 
-    model_type = str
+    model_type: Callable[[Any], str] = str
```

Comment:
Right now this is saying that it takes in any parameters and outputs a str, and is defaulting to the builtin function `str()`. Not the clearest syntax.

I don't know if there's a way to do that with a TypeVar since you would have to dynamically bind it to the builtin cast function, seems to me like that would require another if-else block to do easily. 

Plus, for this function, we know that it always returns a string since it's bound to the string cast function, so a TypeVar would actually be incorrect as it's always going to be a string.

Reply:
Oh, ok, there seem to be two approaches how to model the types here:

- what you did, understand `model_type` as the function which parses `Any` -> `<the target type>`, which works since classes `int`, `float`, etc are also callable objects which return an instance of that class
- what I had in mind: `model_type` is the class that values of the given type should have, which also happens to be a callable

However, I'm pretty sure that the output type is not actually `str`, but at this point you cannot be more precise than `Any`. For example, consider when `model_type = int`.

The best way to express all of this might be using generic, cf. https://stackoverflow.com/questions/74885777/how-to-implement-associated-types-in-python-mypy-or-what-to-do-when-wanting-su

Let's not do that now I'd say, but go for the `Callable` approach.

Reply:
So leave as is?